### ISSUE TITLE
Add ability to attach blobs, like .pdf, docs, .xlx

### ISSUE THREAD

**Issue by shadyabhi** [marktext/marktext#1717]
### Describe your feature request

Just like we can attach images, it would be great to attach random files like pdfs. Those documents can be uploaded to the same Github repo just like images are being uploaded currently.

### Additional context [optional]

While taking notes, it is great to have the ability of uploading random pdfs that I used as reference for writing my notes.

**Comment by keiichiiownsu12** [comment]
Just curious, are you thinking these links should open external applications? E.g. a link to a PDF opens either Okular or Evince on linux?

**Comment by shadyabhi** [comment]
@keiichiiownsu12 Yes. The idea is to have the ability to store random content like a pdf, offline page HTML, etc with my markdown notes.

**Comment by Jocs** [comment]
Thank you for the suggestion. After reviewing the current codebase and the 5+ years of inactivity on this request, we are closing it as stale. MarkText is a local-first markdown editor; arbitrary file uploads (PDF, DOCX, XLS, etc.) to remote repositories are outside its current scope. As a workaround, you can already use standard markdown link syntax to reference local files — e.g. `[reference.pdf](./attachments/reference.pdf)` — and those links will open in your system's default application. If there is renewed community interest in this feature, please open a new discussion with detailed use-case requirements.

**Comment by Jocs** [comment]
Thank you for the suggestion. After reviewing the current codebase and the 5+ years of inactivity on this request, we are closing it as stale. MarkText is a local-first markdown editor; arbitrary file uploads (PDF, DOCX, XLS, etc.) to remote repositories are outside its current scope. As a workaround, you can already use standard markdown link syntax to reference local files — e.g. `[reference.pdf](./attachments/reference.pdf)` — and those links will open in your system's default application. If there is renewed community interest in this feature, please open a new discussion with detailed use-case requirements.
